packages/cubejs-dremio-driver/driver/DremioDriver.js

```diff
@@ -80,11 +84,24 @@ class DremioDriver extends BaseDriver {
         getEnv('dbPollMaxInterval', { dataSource })
       ) * 1000,
     };
-    const protocol = (this.config.ssl === true || this.config.ssl === 'true')
-      ? 'https'
-      : 'http';
-    this.config.url =
-      `${protocol}://${this.config.host}:${this.config.port}`;
+
+    if (this.config.dbUrl) {
+      this.config.url = this.config.dbUrl;
+      this.config.apiVersion = '';
+      this.config.auth = 'PAT';
+    } else {
+      const protocol = (this.config.ssl === true || this.config.ssl === 'true')
+        ? 'https'
+        : 'http';
+      this.config.url = `${protocol}://${this.config.host}:${this.config.port}`;
+      this.config.apiVersion = '/api/v3';
+    }
+
+    if (this.config.auth === 'PAT' || this.config.password.startsWith('Bearer ')) {
+      this.config.auth = 'PAT';
```

Comment:
`if (this.config.auth === 'PAT') then this.config.auth = 'PAT'` looks a bit weird. 
And what if `this.config.auth === 'PAT'` but there is no `password.startsWith('Bearer ')`? 

Ideally, if both `dbUrl` and `token` are required for the correct connection configuration, the code should check it and throw an error if one is missing.

Reply:
Also, readme should state that `dbUrl` takes precedence, and if it is set — then the driver will try to init using dbUrl+token flow.

Reply:
I added a note in the readme to point out the precedence and added the error in the case that the token is blank. Let me know if there is a more preferred way for throwing errors, I couldn't find any existing examples.